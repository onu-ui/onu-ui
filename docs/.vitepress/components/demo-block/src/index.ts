import type { ExtractPropTypes } from 'vue'

export const demoProps = {
  github: {
    type: String,
    default: 'https://github.com/onu-ui/onu-ui',
  },
  codeSandBox: {
    type: String,
    default: 'https://codesandbox.io/',
  },
  highlightedCode: {
    type: String,
    default: '',
  },
  code: {
    type: String,
    default: '',
  },
  title: {
    type: String,
    default: '',
  },
  desc: {
    type: String,
    default: '',
  },
  lang: {
    type: String,
    default: 'vue',
  },
  expand: {
    type: Boolean,
    default: false,
  },
} as const

export type ODemoProps = ExtractPropTypes<typeof demoProps>
