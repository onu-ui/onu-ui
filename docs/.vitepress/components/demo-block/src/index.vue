<script lang='ts' setup name="demo-block">
import { demoProps } from './index'

const props = defineProps(demoProps)

const decodedHighlightedCode = computed(() =>
  decodeURIComponent(props.highlightedCode),
)
const { copy, copied } = useClipboard({ source: decodeURIComponent(props.code) })
const [value, toggle] = useToggle()
</script>

<template>
  <ClientOnly>
    <div v-bind="$attrs" class="mt-6">
      <div class="o-demo_wrapper vp-raw bg">
        <slot />
      </div>
      <div class="relative">
        <div class="o-demo_actions">
          <a class="o-demo_action_item" group :href="codeSandBox" target="_blank">
            <div class="o-demo_action_icon i-carbon:chemistry" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              Open on CodeSandBox
            </div>
          </a>
          <a class="o-demo_action_item" group :href="github" target="_blank">
            <div class="o-demo_action_icon i-carbon-logo-github" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              Edit on GitHub
            </div>
          </a>
          <a class="o-demo_action_item" group @click="copy()">
            <div class="o-demo_action_icon i-carbon:copy" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              {{ copied ? 'Copied' : 'Copy code' }}
            </div>
          </a>
          <a class="o-demo_action_item" group @click="toggle()">
            <div class="o-demo_action_icon i-carbon:fit-to-width" />
            <div class="o-demo_tooltip" group-hover:opacity-100>
              {{ value ? 'Hidden code' : 'Show code' }}
            </div>
          </a>
        </div>
        <div v-show="value" :class="`language-${lang} extra-class`" v-html="decodedHighlightedCode" />
      </div>
    </div>
  </ClientOnly>
</template>

<!-- <style scoped>
.bg {
      /* background-color: #1c1f24; */
      background: conic-gradient(from 192deg at 46.5% 42.58%, #FA8792 -54.94deg, #561BBE 28.29deg,
          #70E6FB 157.82deg, #FBF8B3 220.83deg, #FA8792 305.06deg, #561BBE 388.29deg);
    }
</style> -->
