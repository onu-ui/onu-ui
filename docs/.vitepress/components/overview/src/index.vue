<script lang="ts" setup>
import { Components } from '@onu-ui/docs/guide/components'
import SubTitle from './SubTitle.vue'
import ComponentCard from './ComponentCard.vue'
</script>

<template>
  <div class="space-y-10">
    <div v-for="(category, cKey) in Components" :key="cKey">
      <SubTitle>
        {{ category.name }}
        <span text-xl>{{ `(${category.components.length})` || '' }}</span>
      </SubTitle>
      <div class="grid md:grid-cols-3  gap-5">
        <ComponentCard v-for="(component, pKey) in category.components" :key="pKey" :component="component" />
      </div>
    </div>
  </div>
</template>
