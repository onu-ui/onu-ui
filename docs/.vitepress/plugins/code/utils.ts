import { dirname, join, sep } from 'node:path'
import fsExtra from 'fs-extra'
import type { MarkdownRenderer } from 'vitepress'
import { DemoTag } from './constants'
import type { DemoInfos } from './types'

const scriptRE = /<\/script>/
const scriptLangTsRE = /<\s*script[^>]*\blang=['"]ts['"][^>]*/
const scriptSetupRE = /<\s*script[^>]*\bsetup\b[^>]*/
const scriptClientRE = /<\s*script[^>]*\bclient\b[^>]*/

let index = 1
export function getDemoComponent(
  md: MarkdownRenderer,
  env: any,
  { title, desc, path, code, ...props }: DemoInfos,
) {
  const componentName = `DemoComponent${index++}`
  path = normalizePath(path)
  const github = `https://github.com/onu-ui/onu-ui/tree/main/${path.split('/').slice(-4).join('/')}`

  injectImportStatement(env, componentName, path)

  const highlightedCode = md.options.highlight!(code, props.lang || 'vue', '')
  return `
    <${DemoTag}
      code="${encodeURIComponent(code)}"
      highlightedCode="${encodeURIComponent(highlightedCode)}"
      src="${path}"
      title="${title ?? ''}"
      desc="${desc ?? ''}"
      github="${github}"
    >
        <${componentName}></${componentName}>
    </${DemoTag}>
  `.trim()
}

let fenceIndex = 1
export function genDemoByCode(
  md: MarkdownRenderer,
  env: any,
  path: string,
  code: string,
) {
  let demoName = ''
  let demoPath = ''

  while (true) {
    demoName = `demo-${fenceIndex++}.vue`
    demoPath = join(dirname(path), demoName)
    if (!fsExtra.existsSync(demoPath))
      break
  }

  fsExtra.createFileSync(demoPath)
  fsExtra.writeFileSync(demoPath, code)

  return getDemoComponent(md, env, {
    path: demoPath,
    code,
  })
}

function normalizePath(path: string) {
  return path.split(sep).join('/')
}

function injectImportStatement(
  env: any,
  componentName: string,
  path: string,
) {
  const componentRegistStatement
    = `import ${componentName} from '${path}'`.trim()

  if (!env?.sfcBlocks?.scripts)
    env.sfcBlocks.scripts = []
  const tags = env.sfcBlocks.scripts as { content: string }[]

  const isUsingTS
    = tags.findIndex(tag => scriptLangTsRE.test(tag.content)) > -1
  const existingSetupScriptIndex = tags?.findIndex((tag) => {
    return (
      scriptRE.test(tag.content)
      && scriptSetupRE.test(tag.content)
      && !scriptClientRE.test(tag.content)
    )
  })

  if (existingSetupScriptIndex > -1) {
    const tagSrc = tags[existingSetupScriptIndex]
    tags[existingSetupScriptIndex].content = tagSrc.content.replace(
      scriptRE,
      `${componentRegistStatement}

      </script>`,
    )
  }
  else {
    tags.unshift({
      content: `
        <script ${isUsingTS ? 'lang="ts"' : ''} setup >
          ${componentRegistStatement}
        </script>
      `.trim(),
    })
  }
}
