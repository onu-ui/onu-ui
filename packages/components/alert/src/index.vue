<script lang='ts' setup name="O-Alert">
import OIcon from '../../icon/src/index.vue'
import { alertEmits, alertProps } from './props'

const props = defineProps(alertProps)
const emit = defineEmits(alertEmits)
const slots = useSlots()

const visible = ref(true)

const handleClose = (evt: MouseEvent) => {
  visible.value = false
  emit('close', evt)
}

const showIcon = computed(() => props.icon || slots.icon)
const showTitle = computed(() => props.title || slots.title)
</script>

<template>
  <div v-show="visible" :o="o" class="o-alert-base" :class="[light && 'o-alert-light']">
    <!-- icon -->
    <div v-if="showIcon" class="o-alert-icon">
      <slot name="icon">
        <o-icon :name="icon" class="!c-white text-6" />
      </slot>
    </div>
    <!-- content -->
    <div class="o-alert-content" :class="center && 'items-center'">
      <div class="o-alert-header">
        <div v-if="showTitle" class="o-alert-title">
          <slot name="title">
            {{ title }}
          </slot>
        </div>
        <div v-if="closable" class="o-alert-close" @click="handleClose">
          <slot name="close">
            <o-icon v-if="!closeText" name="i-carbon-close" class="!c-white text-4" />
            <span class="text-xs">{{ closeText }}</span>
          </slot>
        </div>
      </div>
      <div class="o-alert-description" :class="showTitle ? 'mt-1' : ''">
        <slot>
          {{ description }}
        </slot>
      </div>
    </div>
  </div>
</template>
