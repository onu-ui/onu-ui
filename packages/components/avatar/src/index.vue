<script lang='ts' setup name="OAvatar">
import { avatarProps } from './props'

const props = defineProps(avatarProps)

const IsBgMode = computed(() => props.mode === 'bg')
</script>

<template>
  <component
    :is="IsBgMode ? 'div' : 'img'"
    class="o-transition o-avatar-base"
    :class="[IsBgMode ? 'o-avatar-mode-bg' : 'o-avatar-mode-img', `o-avatar-${size}`]"
    v-bind="IsBgMode ? { style: { backgroundImage: `url(${src})` } } : { src, alt: 'Avatar' }"
  >
    <slot v-if="IsBgMode" />
  </component>
</template>
