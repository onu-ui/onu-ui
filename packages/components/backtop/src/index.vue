<script setup lang="ts" name="OBacktop">
import { backtopEmits, backtopProps } from './backtop'
import { useBackTop } from './use-backtop'

const props = defineProps(backtopProps)
const emit = defineEmits(backtopEmits)

const { handleClick, visible } = useBackTop(props, emit, 'OBacktop')
const backTopStyle = computed(() => ({
  right: `${props.right}px`,
  bottom: `${props.bottom}px`,
}))
</script>

<template>
  <transition>
    <div
      v-if="visible"
      :style="backTopStyle"
      class="fixed bg-blue-500 shadow-lg shadow-context:50 w-40px h-40px rounded-1/2 flex justify-center items-center text-20px cursor-pointer z-5 hover:bg-blue-400"
      @click.stop="handleClick"
    >
      <slot>
        <div i-carbon-up-to-top class="!c-white" />
      </slot>
    </div>
  </transition>
</template>

<style scoped>
.v-enter-active,
.v-leave-active {
  transition: opacity 0.5s ease;
}

.v-enter-from,
.v-leave-to {
  opacity: 0;
}
</style>
