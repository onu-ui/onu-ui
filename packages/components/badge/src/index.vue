<script lang="ts" setup name="OBadge">
import { isNumber } from '@onu-ui/utils'
import { badgeProps } from './props'

const props = defineProps(badgeProps)

const resolveValue = computed(() =>
  props.dot
    ? ''
    : (isNumber(props.value) && props.value > props.max)
        ? `${props.max}+`
        : props.value,
)
</script>

<template>
  <span class="o-badge" o-error v-bind="$attrs">
    <slot />
    <sup
      v-if="resolveValue !== 0 || showZero"
      class="o-badge-sup o-badge-sup-fixed"
      :class="dot && 'o-badge-sup-dot'"
    >{{ resolveValue }}</sup>
  </span>
</template>
