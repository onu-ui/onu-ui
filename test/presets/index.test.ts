import { describe, expect, it } from 'vitest'
import { createGenerator } from 'unocss'
import { presetOnu } from '../../packages/preset/src/index'
import { formatCSSCode } from '../utils'

describe('onu preset', () => {
  const testColor = '#9955FF'

  const uno = createGenerator({
    presets: [presetOnu({ color: testColor })],
  })

  it('preflights', async () => {
    const { css } = await uno.generate('')
    expect(await formatCSSCode(css)).toMatchSnapshot()
  })
})
