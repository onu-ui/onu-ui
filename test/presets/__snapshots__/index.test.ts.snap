// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`onu preset > preflights 1`] = `
"/* layer: imports */
@import url("https://fonts.googleapis.com/css2?family=DM+Sans&family=DM+Sans:wght@400;700&display=swap");
/* layer: preflights */
*,
::before,
::after {
  --un-rotate: 0;
  --un-rotate-x: 0;
  --un-rotate-y: 0;
  --un-rotate-z: 0;
  --un-scale-x: 1;
  --un-scale-y: 1;
  --un-scale-z: 1;
  --un-skew-x: 0;
  --un-skew-y: 0;
  --un-translate-x: 0;
  --un-translate-y: 0;
  --un-translate-z: 0;
  --un-pan-x: ;
  --un-pan-y: ;
  --un-pinch-zoom: ;
  --un-scroll-snap-strictness: proximity;
  --un-ordinal: ;
  --un-slashed-zero: ;
  --un-numeric-figure: ;
  --un-numeric-spacing: ;
  --un-numeric-fraction: ;
  --un-border-spacing-x: 0;
  --un-border-spacing-y: 0;
  --un-ring-offset-shadow: 0 0 rgb(0 0 0 / 0);
  --un-ring-shadow: 0 0 rgb(0 0 0 / 0);
  --un-shadow-inset: ;
  --un-shadow: 0 0 rgb(0 0 0 / 0);
  --un-ring-inset: ;
  --un-ring-offset-width: 0px;
  --un-ring-offset-color: #fff;
  --un-ring-width: 0px;
  --un-ring-color: rgb(147 197 253 / 0.5);
  --un-blur: ;
  --un-brightness: ;
  --un-contrast: ;
  --un-drop-shadow: ;
  --un-grayscale: ;
  --un-hue-rotate: ;
  --un-invert: ;
  --un-saturate: ;
  --un-sepia: ;
  --un-backdrop-blur: ;
  --un-backdrop-brightness: ;
  --un-backdrop-contrast: ;
  --un-backdrop-grayscale: ;
  --un-backdrop-hue-rotate: ;
  --un-backdrop-invert: ;
  --un-backdrop-opacity: ;
  --un-backdrop-saturate: ;
  --un-backdrop-sepia: ;
}
::backdrop {
  --un-rotate: 0;
  --un-rotate-x: 0;
  --un-rotate-y: 0;
  --un-rotate-z: 0;
  --un-scale-x: 1;
  --un-scale-y: 1;
  --un-scale-z: 1;
  --un-skew-x: 0;
  --un-skew-y: 0;
  --un-translate-x: 0;
  --un-translate-y: 0;
  --un-translate-z: 0;
  --un-pan-x: ;
  --un-pan-y: ;
  --un-pinch-zoom: ;
  --un-scroll-snap-strictness: proximity;
  --un-ordinal: ;
  --un-slashed-zero: ;
  --un-numeric-figure: ;
  --un-numeric-spacing: ;
  --un-numeric-fraction: ;
  --un-border-spacing-x: 0;
  --un-border-spacing-y: 0;
  --un-ring-offset-shadow: 0 0 rgb(0 0 0 / 0);
  --un-ring-shadow: 0 0 rgb(0 0 0 / 0);
  --un-shadow-inset: ;
  --un-shadow: 0 0 rgb(0 0 0 / 0);
  --un-ring-inset: ;
  --un-ring-offset-width: 0px;
  --un-ring-offset-color: #fff;
  --un-ring-width: 0px;
  --un-ring-color: rgb(147 197 253 / 0.5);
  --un-blur: ;
  --un-brightness: ;
  --un-contrast: ;
  --un-drop-shadow: ;
  --un-grayscale: ;
  --un-hue-rotate: ;
  --un-invert: ;
  --un-saturate: ;
  --un-sepia: ;
  --un-backdrop-blur: ;
  --un-backdrop-brightness: ;
  --un-backdrop-contrast: ;
  --un-backdrop-grayscale: ;
  --un-backdrop-hue-rotate: ;
  --un-backdrop-invert: ;
  --un-backdrop-opacity: ;
  --un-backdrop-saturate: ;
  --un-backdrop-sepia: ;
}
/* layer: @onu-ui/preset-reset */
*,
::before,
::after {
  box-sizing: border-box;
  border-width: 0;
  border-style: solid;
  border-color: var(--un-default-border-color, #e5e7eb);
}
html,
:host {
  line-height: 1.5;
  -webkit-text-size-adjust: 100%;
  -moz-tab-size: 4;
  tab-size: 4;
  font-family: ui-sans-serif, system-ui, sans-serif, "Apple Color Emoji",
    "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  font-feature-settings: normal;
  font-variation-settings: normal;
  -webkit-tap-highlight-color: transparent;
}
body {
  margin: 0;
  line-height: inherit;
}
hr {
  height: 0;
  color: inherit;
  border-top-width: 1px;
}
abbr:where([title]) {
  text-decoration: underline dotted;
}
h1,
h2,
h3,
h4,
h5,
h6 {
  font-size: inherit;
  font-weight: inherit;
}
a {
  color: inherit;
  text-decoration: inherit;
}
b,
strong {
  font-weight: bolder;
}
code,
kbd,
samp,
pre {
  font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas,
    "Liberation Mono", "Courier New", monospace;
  font-feature-settings: normal;
  font-variation-settings: normal;
  font-size: 1em;
}
small {
  font-size: 80%;
}
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}
sub {
  bottom: -0.25em;
}
sup {
  top: -0.5em;
}
table {
  text-indent: 0;
  border-color: inherit;
  border-collapse: collapse;
}
button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-feature-settings: inherit;
  font-variation-settings: inherit;
  font-size: 100%;
  font-weight: inherit;
  line-height: inherit;
  color: inherit;
  margin: 0;
  padding: 0;
}
button,
select {
  text-transform: none;
}
button,
[type="button"],
[type="reset"],
[type="submit"] {
  -webkit-appearance: button;
  background-image: none;
}
:-moz-focusring {
  outline: auto;
}
:-moz-ui-invalid {
  box-shadow: none;
}
progress {
  vertical-align: baseline;
}
::-webkit-inner-spin-button,
::-webkit-outer-spin-button {
  height: auto;
}
[type="search"] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}
::-webkit-search-decoration {
  -webkit-appearance: none;
}
::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}
summary {
  display: list-item;
}
blockquote,
dl,
dd,
h1,
h2,
h3,
h4,
h5,
h6,
hr,
figure,
p,
pre {
  margin: 0;
}
fieldset {
  margin: 0;
  padding: 0;
}
legend {
  padding: 0;
}
ol,
ul,
menu {
  list-style: none;
  margin: 0;
  padding: 0;
}
dialog {
  padding: 0;
}
textarea {
  resize: vertical;
}
input::placeholder,
textarea::placeholder {
  opacity: 1;
  color: #9ca3af;
}
button,
[role="button"] {
  cursor: pointer;
}
:disabled {
  cursor: default;
}
img,
svg,
video,
canvas,
audio,
iframe,
embed,
object {
  display: block;
  vertical-align: middle;
}
img,
video {
  max-width: 100%;
  height: auto;
}
[hidden] {
  display: none;
}
/* layer: @onu-ui/preset-theme */
:root {
  color-scheme: light dark;
  --onu-color-50: 258 100 97;
  --onu-color-100: 256 100 95;
  --onu-color-200: 255 100 92;
  --onu-color-300: 258 100 85;
  --onu-color-400: 261 100 76;
  --onu-color-500: 264 100 67;
  --onu-color-600: 268 93 58;
  --onu-color-700: 269 78 50;
  --onu-color-800: 269 78 42;
  --onu-color-900: 269 75 35;
  --onu-color-950: 267 81 23;
  --onu-color-DEFAULT: var(--onu-color-500);
  --onu-color-text: var(--onu-color-100);
  --onu-color-text-invert: var(--onu-color-950);
}
.dark {
  --onu-color-DEFAULT: var(--onu-color-600);
  --onu-color-text: var(--onu-color-950);
  --onu-color-text-invert: var(--onu-color-100);
}
::selection {
  color: hsl(var(--onu-color-DEFAULT) / 0.99) !important;
  background-color: hsl(var(--onu-color-text) / 0.99) !important;
}
"
`;
